controllers/quay/quayregistry_controller.go

```diff
@@ -358,21 +365,19 @@ func (r *QuayRegistryReconciler) Reconcile(ctx context.Context, req ctrl.Request
 						LastUpdateTime:     metav1.Now(),
 						LastTransitionTime: metav1.Now(),
 					}
-					updatedQuay.Status.Conditions = v1.SetCondition(updatedQuay.Status.Conditions, condition)
-					updatedQuay.Status.CurrentVersion = v1.QuayVersionCurrent
-					r.EventRecorder.Event(updatedQuay, corev1.EventTypeNormal, string(v1.ConditionReasonHealthChecksPassing), msg)
+					qcopy.Status.Conditions = v1.SetCondition(qcopy.Status.Conditions, condition)
+					qcopy.Status.CurrentVersion = v1.QuayVersionCurrent
+					r.EventRecorder.Event(qcopy, corev1.EventTypeNormal, string(v1.ConditionReasonHealthChecksPassing), msg)
 
-					if err = r.Client.Status().Update(ctx, updatedQuay); err != nil {
+					if err = r.Client.Status().Update(ctx, qcopy); err != nil {
 						log.Error(err, "could not update QuayRegistry status with current version")
-
-						return true, err
+						return false, nil
```

Comment:
Why is the error not returned anymore, and how will the controller handle that change?

Reply:
The idea is to make the go routine resilient to errors by simply logging them instead of returning them. If we return an error the go routine will stop trying and this is what we are trying to avoid. As a side note, this go routine must go away because it is wrong.